### PR TITLE
should not fail to resolve hosts in hosts file when that file has CRLF EOF

### PR DESCRIPTION
**Pull request by fluffynuts**
This should fix the issue raised against got: https://github.com/sindresorhus/got/issues/1197

Windows hosts files can have CRLF EOL characters and the existing code doesn't cater for that, instead failing to resolve hosts in the hosts file.

### CODE REVIEW

File: tests/test.js [hunk 1]
```diff
@@ -782,6 +783,23 @@ test.serial('double tick() has no effect', t => {
...
+			return `${line}\r`;
+		}
+
+		return line;
+	});
+	if (!altered.find(line => line.includes('has-windows-newline'))) {
+		throw new Error('Can\'t find "has-windows-newline" hosts entry');
+	}
+
+	await fs.writeFile(hostFile, Buffer.from(altered.join('\n')));
```
Comment (sindresorhus) [inline tests/test.js:801]: This is a lot of extra logic. I would move this to a separate test.

File: tests/test.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const path = require('path');
 const test = require('ava');
 const Keyv = require('keyv');
 const proxyquire = require('proxyquire');
+const fs = require('fs').promises;
```
Comment (sindresorhus) [inline tests/test.js:9]:
```suggestion
const {promises: fs} = require('fs');
```

**Comment by szmarczak** [comment]
Fixed in #26 - this just needs a test. I would prefer a separate file called `clrfHosts.txt`.

**Comment by szmarczak** [comment]
Added tests 615412c07516447babf075a2f960c6ca770f82b4, it just needs tests for EOF.

**Comment by szmarczak** [comment]
> it just needs tests for EOF

Actually it's done already.

Thanks for the PR @fluffynuts :raised_hands: You gave me an important hint, I really appreciate it!

**Comment by fluffynuts** [comment]
I wasn't sure whether to add the eol test to the existing one or make a new test. In my own code, I would have added a new test, but the existing one looked quite comprehensive, so I assumed that was more the style of the project - some people prefer, for example, one test that does a lot of stuff but keeps the logic of a feature in one place. Personally, I prefer lots of small, specific tests. When making a PR, I really try to stick with the original style of the upstream project.

**Pull request closed without merging**